google-cloud-storage/src/test/java/com/google/cloud/storage/StorageImplMockitoTest.java

```diff
@@ -458,6 +459,191 @@ public void testCreateBucketFailure() {
+    String selector = (String) capturedOptions.getValue().get(BLOB_GET_FIELDS.getRpcOption());
+    assertTrue(selector.contains("name"));
+    assertTrue(selector.contains("location"));
+    assertTrue(selector.contains("acl"));
+    assertEquals(17, selector.length());
+    assertEquals(BUCKET_INFO1.getName(), bucket.getName());
+  }
+
+  @Test
+  public void testGetBucketWithEmptyFields() {
+    ArgumentCaptor<Map<StorageRpc.Option, Object>> capturedOptions =
+        ArgumentCaptor.forClass(Map.class);
+    doReturn(BUCKET_INFO1.toPb())
+        .doThrow(UNEXPECTED_CALL_EXCEPTION)
+        .when(storageRpcMock)
+        .get(Mockito.eq(BucketInfo.of(BUCKET_NAME1).toPb()), capturedOptions.capture());
+    initializeService();
+    Bucket bucket = storage.get(BUCKET_NAME1, BUCKET_GET_METAGENERATION, BUCKET_GET_EMPTY_FIELDS);
+    assertEquals(
+        BUCKET_GET_METAGENERATION.getValue(),
+        capturedOptions.getValue().get(BUCKET_GET_METAGENERATION.getRpcOption()));
+    String selector = (String) capturedOptions.getValue().get(BLOB_GET_FIELDS.getRpcOption());
+    assertTrue(selector.contains("name"));
+    assertEquals(4, selector.length());
+    assertEquals(BUCKET_INFO1.getName(), bucket.getName());
+  }
+
+  @Test
+  public void testGetBucketFailure() {
+    doThrow(STORAGE_FAILURE)
+        .when(storageRpcMock)
+        .get(BucketInfo.of(BUCKET_NAME1).toPb(), EMPTY_RPC_OPTIONS);
+    initializeService();
+    try {
+      storage.get(BUCKET_NAME1);
+      fail();
+    } catch (StorageException e) {
+      assertEquals(STORAGE_FAILURE, e.getCause());
+    }
+  }
+
+  @Test
+  public void testGetBlob() {
+    doReturn(BLOB_INFO1.toPb())
+        .doThrow(UNEXPECTED_CALL_EXCEPTION)
+        .when(storageRpcMock)
+        .get(BlobId.of(BUCKET_NAME1, BLOB_NAME1).toPb(), EMPTY_RPC_OPTIONS);
+    initializeService();
+    Blob blob = storage.get(BUCKET_NAME1, BLOB_NAME1);
+    assertEquals(expectedBlob1, blob);
+  }
+
+  @Test
+  public void testGetBlobWithOptions() {
+    doReturn(BLOB_INFO1.toPb())
+        .doThrow(UNEXPECTED_CALL_EXCEPTION)
+        .when(storageRpcMock)
+        .get(BlobId.of(BUCKET_NAME1, BLOB_NAME1).toPb(), BLOB_GET_OPTIONS);
+    initializeService();
+    Blob blob = storage.get(BUCKET_NAME1, BLOB_NAME1, BLOB_GET_METAGENERATION, BLOB_GET_GENERATION);
+    assertEquals(expectedBlob1, blob);
+  }
+
+  @Test
+  public void testGetBlobWithOptionsFromBlobId() {
+    doReturn(BLOB_INFO1.toPb())
+        .doThrow(UNEXPECTED_CALL_EXCEPTION)
+        .when(storageRpcMock)
+        .get(BLOB_INFO1.getBlobId().toPb(), BLOB_GET_OPTIONS);
+    initializeService();
+    Blob blob =
+        storage.get(
+            BLOB_INFO1.getBlobId(), BLOB_GET_METAGENERATION, BLOB_GET_GENERATION_FROM_BLOB_ID);
+    assertEquals(expectedBlob1, blob);
+  }
+
+  @Test
+  public void testGetBlobWithSelectedFields() {
+    ArgumentCaptor<Map<StorageRpc.Option, Object>> capturedOptions =
```

Comment:
you can define argumentcapture at global level so you don't need to declare in every test case

Reply:
Thanks for the suggestion. argumentCapture is used not in all cases, but in a few ones. If we declare it globally we have to clean it up in setUp() method which is runnig before each test.  Other tests use other captures, so I think I will leave this as is.